src/journal/journal-file.h

```diff
@@ -89,6 +90,7 @@ typedef struct JournalFile {
         bool defrag_on_close:1;
         bool close_fd:1;
         bool archive:1;
+        bool eio:1; /* permanently set when -EIO was ever returned from fsync() */
```

Comment:
That's fair, I had just looked at the `fsync(2)` man page ERRORS section:
```
ERRORS
       EBADF  fd is not a valid open file descriptor.

       EIO    An error occurred during synchronization.

       EROFS, EINVAL
              fd is bound to a special file (e.g., a pipe, FIFO, or socket) which
              does not support synchronization.
```

EIO seemed like the important one.

It would be trivial to change the `eio` bool to something like `int fsync_error` and treat it as relevant when negative.

Reply:
@poettering pushed a commit undoing the EIO discrimination